packages/nocodb/src/version-upgrader/ncXcdbLTARIndexUpgrader.ts

```diff
@@ -0,0 +1,165 @@
+      case RelationTypes.BELONGS_TO:
+        {
+          // const parentCol = await colOptions.getParentColumn(ncMeta);
+          const childCol = await colOptions.getChildColumn(ncMeta);
+
+          // const parentModel = await parentCol.getModel(ncMeta);
+          const childModel = await childCol.getModel(ncMeta);
+
+          // check index already exists or not
+          const indexExists = indexes.find((index) => {
+            return index.cn === childCol.column_name;
+          });
+
+          if (indexExists) {
+            continue;
+          }
+
+          logger.log(`Creating index for column '${childCol.column_name}'`);
+          // create a new index for the column
+          const indexArgs = {
+            columns: [childCol.column_name],
+            tn: childModel.table_name,
+            non_unique: true,
+          };
+          await sqlClient.indexCreate(indexArgs);
+        }
+        break;
+    }
+  }
+}
+
+// An upgrader for adding missing index for LTAR relations in XCDB bases
+async function upgradeBaseRelations({
+  ncMeta,
+  base,
+}: {
+  ncMeta: MetaService;
+  base: Base;
+}) {
+  // skip deleted projects
+  if ((await base.getProject(ncMeta)).deleted) return;
+
+  const sqlClient = await NcConnectionMgrv2.getSqlClient(base, ncMeta.knex);
+
+  // get models for the base
+  const models = await ncMeta.metaList2(null, base.id, MetaTable.MODELS);
+
+  // get all columns and filter out relations and create index if not exists
+  for (const model of models) {
+    logger.log(`Upgrading model '${model.title}'`);
+
+    logger.log(`Fetching index list of model '${model.title}'`);
+    const indexes = await sqlClient.indexList({
+      tn: model.table_name,
+    });
+    const indexList = indexes.data?.list ?? [];
+
+    // exclude composite indexes
+    const filteredIndexList = indexList.filter((indexObj, i) => {
+      return indexList.every((indexObj2, j) => {
+        if (i === j) return true;
+        return indexObj2.key_name !== indexObj.key_name;
+      });
+    });
+
+    await upgradeModelRelationsIndex({
+      ncMeta,
+      model: new Model(model),
+      sqlClient,
+      indexes: filteredIndexList,
+    });
+    logger.log(`Upgraded model '${model.title}'`);
+  }
+}
+
+// Add missing index for LTAR relations
+export default async function ({ ncMeta }: NcUpgraderCtx) {
+  logger.log(
+    'Starting upgrade for LTAR relations in XCDB bases to add missing index',
+  );
+
+  // get all xcdb bases
+  const bases = await ncMeta.metaList2(null, null, MetaTable.BASES, {
+    condition: {
+      is_meta: 1,
+    },
+    orderBy: {},
+  });
+
+  if (!bases.length) return;
+
+  // iterate and upgrade each base
+  for (const base of bases) {
+    // skip if not pg, since for other db we don't need to upgrade
+    if (ncMeta.knex.clientType() !== 'pg') {
+      continue;
+    }
+
+    logger.log(`Upgrading base '${base.name}'`);
```

Comment:
base.name seem to be undefined in all cases maybe we can do alias ?? id in order to get some information